src/providers/twitch/api/Helix.hpp

```diff
@@ -0,0 +1,197 @@
+#pragma once
+
+#include "common/NetworkRequest.hpp"
+
+#include <QString>
+#include <QStringList>
+#include <QUrl>
+#include <QUrlQuery>
+#include <boost/noncopyable.hpp>
+#include <boost/optional.hpp>
+
+#include <functional>
+#include <vector>
+
+namespace chatterino {
+
+using HelixFailureCallback = std::function<void()>;
+template <typename... T>
+using ResultCallback = std::function<void(T...)>;
+
+struct HelixUser {
+    const QString id;
```

Comment:
The idea of `const` being used within the class is to explicitly not allow any modification of the members (making it immutable). The only other place where it would have the same effect would be in the result callback in the calling code but the user could just as easily omit the `const` flag and have the object mutable. `getHelix()->getUserByName("pajlada", [](HelixUser user) { user.id="xd"; }, []{});`
I'm fine to drop the `const` members, I just want to verify that this is what you meant. `const` for each member might be too much hand-holding

Reply:
Why do we have to limit the modification to begin with? There is only a single callback, which has exclusive control over it.